File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/FireExtinguisher.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

import org.rexcellentgames.burningknight.assets.Locale;

public class FireExtinguisher extends Equippable {
	{
		name = Locale.get("fire_extinguisher");
		description = Locale.get("fire_extinguisher_desc");
		sprite = "item-fire_extinguisher";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.fireResist += 1;
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.fireResist -= 1;
	}

	@Override
	public boolean canBeUpgraded() {
		return false;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/FireExtinguisher.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class FireExtinguisher : Equippable() {
	init {
		name = Locale.get("fire_extinguisher")
		description = Locale.get("fire_extinguisher_desc")
		sprite = "item-fire_extinguisher"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.fireResist += 1
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.fireResist -= 1
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}